Integrations/EDL/EDL.py

```diff
@@ -102,16 +126,32 @@ def find_indicators_to_limit(indicator_query: str, limit: int,
         indicator_query (str): Query that determines which indicators to include in
             the EDL (Cortex XSOAR indicator query syntax)
         limit (int): The maximum number of indicators to include in the EDL
+        offset (int): The starting index from which to fetch incidents
         panos_compatible (bool): Whether to make the indicators PANOS compatible or not
         url_port_stripping (bool): Whether to strip the port from URL indicators (if a port is present) or not
 
     Returns:
         list: The IoCs list up until the amount set by 'limit'
     """
-    iocs, _ = find_indicators_to_limit_loop(indicator_query, limit,
+    if offset:
+        next_page = int(offset / PAGE_SIZE)
+
+        # set the offset from the starting page
+        offset_in_page = offset - (PAGE_SIZE * next_page)
+
+    else:
+        next_page = 0
+        offset_in_page = 0
+
+    iocs, _ = find_indicators_to_limit_loop(indicator_query, limit, next_page=next_page,
                                             panos_compatible=panos_compatible,
                                             url_port_stripping=url_port_stripping)
-    return iocs[:limit]
+
+    # if offset in page is bigger than the amount of results returned return empty list
+    if len(iocs) <= offset_in_page:
+        return []
+
+    return iocs[offset_in_page:limit + offset_in_page]
```

Comment:
Can't we have here a situation when `len(iocs) < limit + offset_in_page - 1`?

Reply:
We can but this is ok - in slicing if the upper limit is longer than the length of the array the full array is returned (https://v4.software-carpentry.org/python/slice.html)